### PR TITLE
Add kube-state-metrics example

### PR DESCRIPTION
**Pull request by fabxc**

### CODE REVIEW

**Comment by pintohutch** [comment]
LGTM % worth folding in https://github.com/GoogleCloudPlatform/prometheus-engine/blob/f8a719bba5ae91321ae4a783c5954bc94675f593/cmd/operator/deploy/kube-state-metrics/horizontal_pod_autoscaler.yaml and deleting `cmd/operator/deploy/kube-state-metrics` in favor of this?

**Review by pintohutch — APPROVED** [review]
LGTM % worth folding in https://github.com/GoogleCloudPlatform/prometheus-engine/blob/f8a719bba5ae91321ae4a783c5954bc94675f593/cmd/operator/deploy/kube-state-metrics/horizontal_pod_autoscaler.yaml and deleting cmd/operator/deploy/kube-state-metrics in favor of this?

File: examples/kube-state-metrics.yaml [hunk 1]
```diff
@@ -0,0 +1,287 @@
+# Copyright 2021 Google LLC
```
Comment (TheSpiritXIII) [inline examples/kube-state-metrics.yaml:1]: nit: we're in 2022

Comment (pintohutch) [reply examples/kube-state-metrics.yaml:1]: Yeah - let's fix the header in a separate PR

**Pull request merged**